src/Text/Parsing/Parser.purs

```diff
@@ -136,3 +137,9 @@ fail message = failWithPosition message =<< position
 -- | Fail with a message and a position.
 failWithPosition :: forall m s a. Monad m => String -> Position -> ParserT s m a
 failWithPosition message pos = throwError (ParseError message pos)
+
+-- | If parsing fails inside this labelled context, then prepend the `String`
+-- | to the error `String` in the `ParseError`.
+label :: forall m s a. Monad m => String -> ParserT s m a -> ParserT s m a
+label messagePrefix p = catchError p $ \ (ParseError message pos) ->
```

Comment:
Is there value in appending with a space by default, or should that be left to the user?

Reply:
Yeah good question. I say leave it to the user, who might reasonably choose to append `" "` or `"\n"` or something else.